features/storage/blockdevice/MBRBlockDevice.h

```diff
@@ -52,21 +52,25 @@ class MBRBlockDevice : public BlockDevice {
      *
      *  @param bd       Block device to partition
      *  @param part     Partition to use, 1-4
-     *  @param type     8-bit partition type to identify partition contents
+     *  @param type     8-bit partition type or ID to identify partition contents
      *  @param start    Start block address to map to block 0 of partition.
      *                  Negative addresses are calculated from the end of the
      *                  underlying block devices. Block 0 is implicitly ignored
      *                  from the range to store the MBR.
      *  @return         0 on success or a negative error code on failure.
      *  @note This is the same as partition(bd, part, type, start, bd->size())
+     *  @note 8-bit partition type or ID is intended to specify the filesystem partition
+     *        contains and/or to flag special access methods used to access these partitions
+     *        for example like CHS mapping, LBA access etc.,
+     *        The list of Partition types or IDS refer https://en.wikipedia.org/wiki/Partition_type#List_of_partition_IDs
      */
     static int partition(BlockDevice *bd, int part, uint8_t type, bd_addr_t start);
 
     /** Format the MBR to contain the following partition
      *
      *  @param bd       Block device to partition
      *  @param part     Partition to use, 1-4
-     *  @param type     8-bit partition type to identify partition contents
+     *  @param type     8-bit partition type or ID to identify partition contents
```

Comment:
```suggestion
     *  @param type     8-bit partition type or ID to identify partition contents.
     *                  Refer to https://en.wikipedia.org/wiki/Partition_type#List_of_partition_IDs 
```

Reply:
As per my comment a few comments above, I personally don't think linking to a long external list is a good idea. The user should know what partition (e.g. FAT, EXT4, etc.) they intend to use and set the partition type according to the FS's documentation.

Reply:
@LDong-Arm  The issue reported for missing this partition type information and request to add

Reply:
Yes, we missed an explanation for the "type" parameter. I think what we need to add is what a type is (as you added in this PR), and how to find out the value (e.g. from the file system's documentation). So users of this API will know how to find it out.